datatypes/src/jvmTest/kotlin/PublicKeyTest.kt

```diff
@@ -50,6 +51,16 @@ class PublicKeyTest : FreeSpec({
                 CryptoPublicKey.decodeFromTlv(Asn1Element.parse(own.derEncoded) as Asn1Sequence) shouldBe own
             }
         }
+
+        "Equality tests" {
+            val keyPair = KeyPairGenerator.getInstance("EC").also { it.initialize(256) }.genKeyPair()
+            val pubKey1 = CryptoPublicKey.derDecode(keyPair.public.encoded)
+            val pubKey2 = CryptoPublicKey.derDecode(keyPair.public.encoded)
+
+            pubKey1.hashCode() shouldBe pubKey2.hashCode()
```

Comment:
why bother comparing the hash codes?

Reply:
because `equals` and `hashCode` are not always called together